### ISSUE TITLE
Usability tweaks

### ISSUE THREAD

**Issue by mpickering** [mpickering/ghcide#2]
* Need to cancel duplicated events in the priority queue, typing can lead to a lot of propagate events queuing up which can lead to some blocking.
* Need to put each Propagate individually into the queue rather than as one action as it can cause 
the batch to take a long time
* Only propagate when the module is saved? This could be a nice UI.

**Comment by mpickering** [comment]
Now the queue only contains unique events and changes are only propagated when you save.

There are still lots of knobs to tweak.

**Comment by pepeiborra** [comment]
FWIW I think that Shake will automatically ignore duplicate actions passed in a single call to `shakeRun`, so as long as the Shake queue is flushed entirely (drop priorities), duplicates should not matter.

**Comment by mpickering** [comment]
This is true, but also from a usability perspective you get a lot of redundant diagnostic logging if you don't deduplicate the requests before they get sent to shake so I think it makes sense anyway.
